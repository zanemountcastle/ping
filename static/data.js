export const feeds = [
  {
    "id": "111",
    "title": "There's food in the Baraha.",
    "organization": "NYUAD",
    "color": "#03C7F5",
    "status": "On",
    "lastActive": "one minute ago",
  },
  {
    "id": "112",
    "title": "A new issue of The Gazelle is released.",
    "organization": "The Gazelle",
    "color": "#A327FD",
    "status": "Off",
    "lastActive": "three days ago",
  },
  {
    "id": "113",
    "title": "The A6 hammock is free.",
    "organization": "ResEd",
    "color": "#628395",
    "status": "On",
    "lastActive": "four hours ago",
  },
  {
    "id": "114",
    "title": "The Saints score a touchdown.",
    "organization": "The Saints",
    "color": "#F0CF65",
    "status": "On",
    "lastActive": "one minute ago",
  },
  {
    "id": "115",
    "title": "My coffee is ready.",
    "organization": "Blacksmith Coffee",
    "color": "#40403F",
    "status": "On",
    "lastActive": "one minute ago",
  }
];
