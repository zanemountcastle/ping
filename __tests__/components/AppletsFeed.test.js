import React from 'react';
import AppletFeed from '../../src/components/AppletsFeed';
import { feeds } from '../../static/data.js';

jest.useFakeTimers();

import renderer from 'react-test-renderer';

describe('applet preview', () => {

  it('renders without crashing', () => {
    const rendered = renderer.create(<AppletFeed feeds={feeds} />).toJSON();
    expect(rendered).toBeTruthy();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<AppletFeed feeds={feeds} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  // it('renders correctly over time', () => {
  //   timerGame();
  //
  //   expect(setTimeout.mock.calls.length).toBe(1);
  //   expect(setTimeout.mock.calls[0][1]).toBe(1000);
  // });
  // it('renders 3 times', () => {
  //   // const timerGame = require('../../src/components/AppletsFeed');
  //   const view = renderer.create(<AppletFeed feeds={feeds} />).toJSON();
  //   jest.advanceTimersByTime(1000); // in jest 21.3.0 advanced timer capability
  //   expect(setTimeout.mock.calls.length).toBe(3);
  //   expect(setTimeout.mock.calls[0][1]).toBe(1000);
  // });

});
