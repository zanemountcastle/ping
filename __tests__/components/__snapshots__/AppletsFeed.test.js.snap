// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applet preview renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <ActivityIndicator
    animating={true}
    color="#999999"
    hidesWhenStopped={true}
    size="small"
  />
</View>
`;
